contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -675,3 +708,36 @@ tsql_find_collation_internal(const char *collation_name)
 	return (*collation_callbacks_ptr->find_collation_internal)(collation_name);
 }
 
+bool
+expr_contains_ilike_and_ci_collation_wrapper(Node *expr, bool check_for_ci_as_collation)
+{
+	List 		*queue;
+	ListCell 	*lc = NULL;
+	bool 		expr_contains_ilike_and_ci_as = false;
+	
+	if(expr == NULL)
+		return false;
+	queue = NIL;
+	queue = list_make1(expr);
+
+	while(list_length(queue) > 0)
+	{
+		Node *predicate = (Node *) linitial(queue);
+		queue = list_delete_first(queue);
+		
+		if(IsA(predicate, OpExpr))
+		{
+			/* Initialize collation callbacks */
+			init_and_check_collation_callbacks();
+			expr_contains_ilike_and_ci_as = (*collation_callbacks_ptr->expr_contains_ilike_and_ci_as_coll)(predicate, true);
+			if (expr_contains_ilike_and_ci_as)
+				break;
+		}
+		else if (IsA(predicate, BoolExpr))
+		{
+			BoolExpr   *boolexpr = (BoolExpr *) predicate;			
+			queue = list_concat(queue, boolexpr->args);
+		}
```

Comment:
Do we have a test case for this?

Reply:
I can write a test case to exercise this code, but the function will return false. It will return true only for psql syntax. 

However, I tried a simple test and found a bug in the LIKE to ILIKE transformation logic. The following table definition:
```
CREATE TABLE tbl3(
     a char(20) COLLATE bbf_unicode_cp1_ci_as,
     CHECK (NOT a LIKE '11%' AND a LIKE 'ab%')
)

```
is pg dumped as:
```
CREATE TABLE master_dbo.tbl3 (
    a sys.bpchar(20) COLLATE sys.bbf_unicode_cp1_ci_as,
    CONSTRAINT tbl3_a_check CHECK (((NOT ((a)::text ~~ '11%'::text)) AND (((a)::text ~~* 'ab%'::text) AND (((a)::text >= 'ab'::text) AND ((a)::text < ('ab'::text || '￿'::text COLLATE sys.bbf_unicode_cp1_ci_as))))))
)
```
(note that the `a LIKE '11%'` expression has not been converted into an ILIKE expression.

If it is okay with you, then I would like to file a separate bug for this and work on it separately from this PR since it has not come up yet. Adding the test will be pending this new fix.

Reply:
okay, makes sense.